sdc_client/base_api_client.py

```diff
@@ -0,0 +1,131 @@
+        return self.session.get(self._build_url('system', 'logs'), params=params)
+
+    def get_pipeline(self, pipeline_id: str):
+        self.logger.info(f'Get pipeline `{pipeline_id}`')
+        return self.session.get(self._build_url('pipeline', pipeline_id))
+
+    def get_pipeline_status(self, pipeline_id: str):
+        self.logger.info(f'Get pipeline status `{pipeline_id}`')
+        return self.session.get(self._build_url('pipeline', pipeline_id, 'status'))
+
+    def get_pipeline_history(self, pipeline_id: str):
+        self.logger.info(f'Get pipeline history `{pipeline_id}`')
+        return self.session.get(self._build_url('pipeline', pipeline_id, 'history'))
+
+    def get_pipeline_metrics(self, pipeline_id: str):
+        self.logger.info(f'Get pipeline metrics `{pipeline_id}`')
+        return self.session.get(self._build_url('pipeline', pipeline_id, 'metrics'))
+
+    def reset_pipeline(self, pipeline_id: str):
+        self.logger.info(f'Reset pipeline `{pipeline_id}`')
+        return self.session.post(self._build_url('pipeline', pipeline_id, 'resetOffset'))
+
+    def get_pipeline_offset(self, pipeline_id: str):
+        return self.session.get(self._build_url('pipeline', pipeline_id, 'committedOffsets'))
+
+    def post_pipeline_offset(self, pipeline_id: str, offset: dict):
+        return self.session.post(self._build_url('pipeline', pipeline_id, 'committedOffsets'), json=offset)
+
+    def validate(self, pipeline_id: str):
+        self.logger.info(f'Validate pipeline `{pipeline_id}`')
+        return self.session.get(self._build_url('pipeline', pipeline_id, 'validate'),
+                                params={'timeout': PREVIEW_TIMEOUT})
+
+    def create_preview(self, pipeline_id: str):
+        self.logger.info(f'Create pipeline `{pipeline_id}` preview')
+        return self.session.post(self._build_url('pipeline', pipeline_id, 'preview'),
+                                 params={'timeout': PREVIEW_TIMEOUT})
+
+    def get_preview(self, pipeline_id: str, previewer_id: str):
+        self.logger.info(f'Get preview `{pipeline_id}`')
+        return self.session.get(self._build_url('pipeline', pipeline_id, 'preview', previewer_id))
+
+    def get_preview_status(self, pipeline_id: str, previewer_id: str):
+        self.logger.info(f'Get preview status `{pipeline_id}`')
+        return self.session.get(self._build_url('pipeline', pipeline_id, 'preview', previewer_id, 'status'))
+
+    def get_jmx(self, query: str):
+        return self.session.get(self._build_url('system', 'jmx'), params={'qry': query})
+
+
+
+class ApiClientException(Exception):
+    def __init__(self, message: str, exception_type: str = ''):
+        self.exception_type = exception_type
+        self.message = message
+
+
+class UnauthorizedException(Exception):
+    def __init__(self, message: str):
+        self.message = message
+        self.exception_type = ''
+
+
+class PipelineFreezeException(Exception):
+    pass
```

Comment:
These Exception classes are present in inherited api_client. Do we need them in both?